### PR TITLE
Basic ISIS Server part #1

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: protocols/isis/server/device.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	d := &dev{
+		name:               ifcfg.Name,
+		srv:                srv,
+		passive:            ifcfg.Passive,
+		p2p:                ifcfg.P2P,
+		supportedProtocols: []uint8{packet.NLPIDIPv4, packet.NLPIDIPv6},
+		done:               make(chan struct{}),
+	}
+
+	d.helloMethod = d.receiverRoutine
```
Comment (czerwonk) [inline protocols/isis/server/device.go]: was assigning the same routine to both handlers intended?

File: protocols/isis/server/device.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		return errors.Wrap(err, "Unable to close socket")
+	}
+
+	d.wg.Wait()
+	d.up = false
+	return nil
+}
+
+func (d *dev) receiverRoutine() {
+
```
Comment (czerwonk) [inline protocols/isis/server/device.go]: is leaving this func body empty intended (follow up PR)

File: protocols/isis/server/sys_linux.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	ll := from.(*syscall.SockaddrLinklayer)
+	copy(src[:], ll.Addr[:6])
+
+	return buf[:nBytes], src, nil
+}
+
+func (b *bioSys) sendPacket(pkt []byte, dst [6]byte) error {
+	ll := syscall.SockaddrLinklayer{
+		//Protocol: htons(uint16(len(pkt) + 3)),
```
Comment (czerwonk) [inline protocols/isis/server/sys_linux.go]: if not needed any more: please remove

File: protocols/isis/server/device.go (hunk 2 above)
Comment (taktv6) [reply protocols/isis/server/device.go]: Content to follow in a coming PR

File: protocols/isis/server/device.go (hunk 1 above)
Comment (taktv6) [reply protocols/isis/server/device.go]: Nope. But HelloRoutine did not exist yet. Just created it.

File: protocols/isis/server/sys_linux.go (hunk 3 above)
Comment (taktv6) [reply protocols/isis/server/sys_linux.go]: removed.

**Pull request merged**